synapse/storage/engines/postgres.py

```diff
@@ -38,6 +38,18 @@ def convert_param_style(self, sql):
         return sql.replace("?", "%s")
 
     def on_new_connection(self, db_conn):
+
+        # Get the version of PostgreSQL that we're using. As per the psycopg2
+        # docs: The number is formed by converting the major, minor, and
+        # revision numbers into two-decimal-digit numbers and appending them
+        # together. For example, version 8.1.5 will be returned as 80105.
+        server_version = str(db_conn.server_version)
+        self._version = (
+            int(server_version[:-4], 10),
+            int(server_version[-4:-2], 10),
+            int(server_version[-2:], 10)
+        )
```

Comment:
This feels fragile. Why not do `server_version >= 90500`? The docs also suggest doing `server_version/100 >= 905` to determine feature compatibility based on major versions.